bin/athena.bb

```diff
@@ -29,53 +28,69 @@
        (take-while (every-pred some? (complement z/end?)))
        (some #(when (-> % z/node p?) %))))
 
+(defn tag= [tag]
+  (fn [node]
+    (= tag (:tag node))))
+
+(defn content= [v]
+  (cond
+    (string? v) (fn [node]
+                  (= v (str/join (:content node))))
+    (fn? v)     (fn [node]
+                  (v (str/join (:content node))))
+    :else       (assert false "what did you pass")))
+
 (defn topmost [z]
   (loop [z z]
     (if-let [parent (z/up z)]
       (recur parent)
       z)))
 
 (defn jar-file-name [version]
-   (str "athena-jdbc-" version ".jar"))
+  (str "athena-jdbc-" version ".jar"))
 
 (defn create-new-metadata [version]
-  (let [existing (x/parse (io/reader "https://s3.amazonaws.com/maven-athena/com/metabase/athena-jdbc/maven-metadata.xml"))
-        z (z/xml-zip existing)]
-    (when (find-pred z (every-pred (comp #(= :version %) :tag)
-                                   (comp #(= version %) str/join :content)))
+  (let [existing (x/parse (io/reader "https://s3.amazonaws.com/metabase-maven-downloads/com/metabase/athena-jdbc/maven-metadata.xml"))
+        z        (z/xml-zip existing)]
+    (when (find-pred z (every-pred (tag= :version)
+                                   (content= version)))
       (throw (Exception. "Version already exists in maven-metadata.xml")))
-    (with-open [w (io/writer "maven-metadata.xml")]
-      (-> z
-          (find-pred (comp #(= :versions %) :tag))
-          (z/insert-child (x/element :version {} version))
-          topmost
-          (find-pred (comp #(= :release %) :tag))
-          (z/down)
-          (z/replace version)
-          (z/root)
-          (x/emit w)))))
+    (let [indent (-> z (find-pred (tag= :versions)) z/down first)
+          indent (when (string? indent) indent)
```

Comment:
that's just to get the indentation in the file correctly :-) I've tried to use `x/indent` instead of `x/emit`, but it does some really weird things - maybe it could work if we skip whitespace between tags while reading, but I'm not too sure how to do that. 😁